3.processing_features/scripts/2.pycytominer_singlecell_pipelines.py

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# ## Perform single-cell pycytominer pipelines
+# 
+# Following single-cell curation with cytotable, we create single-cell profiles by applying the following steps:
+# 
+# 1. annotation
+# 2. normalization
+# 3. feature_selection
+# 
+# Additionally, we create bulk profiles following feature selection.
+# We call this "Cameron's Method".
+# 
+# 4. Aggregate (to form bulk, after single-cell processing)
+
+# In[1]:
+
+
+import pathlib
+import yaml
+import pprint
+
+import pandas as pd
+
+from pycytominer import aggregate, annotate, normalize, feature_select
+from pycytominer.cyto_utils import load_profiles, output, infer_cp_features
+
+
+# In[2]:
+
+
+# Set constants
+feature_select_ops = [
+    "variance_threshold",
+    "correlation_threshold",
+    "blocklist",
+]
+
+# Columns to remove prior to single-cell aggregation via cameron's method
+cameron_unwanted_aggregate_cols = {"Object", "Parent", "Site", "Image"}
+
+# Set paths
+output_dir = pathlib.Path("data/single_cell_profiles")
+output_dir.mkdir(exist_ok=True)
+metadata_dir = pathlib.Path("../0.download_data/metadata/")
+
+# load in plate information
+dictionary_path = pathlib.Path("./plate_info_dictionary.yaml")
+with open(dictionary_path) as file:
+    plate_info_dictionary = yaml.load(file, Loader=yaml.FullLoader)
+
+
+# In[3]:
+
+
+# add path to platemaps for each plate 
+for plate in plate_info_dictionary.keys():
+    # since Plate_3_prime has the same platemap as Plate_3,
+    # we need an else statement so that we make sure it adds the 
+    # path that was given to Plate_3
+    if plate != "Plate_3_prime":
+        # match the naming format of the plates to the platemap file
+        plate_info_dictionary[plate]["platemap_path"] = str(
+            pathlib.Path(
+                list(
+                    metadata_dir.rglob(f"platemap_NF1_{plate.replace('_', '').lower()}.csv")
+                )[0]
+            ).resolve(strict=True)
+        )
+    else:
+        plate_info_dictionary["Plate_3_prime"]["platemap_path"] = (
+            plate_info_dictionary["Plate_3"]["platemap_path"]
+        )
+
+# view the dictionary to assess that all info is added correctly
+pprint.pprint(plate_info_dictionary, indent=4)
+
+
+# ## Perform single-cell pycytominer pipeline
+
+# In[4]:
+
+
+for plate, info in plate_info_dictionary.items():
+    print(f"Now performing single-cell pycytominer pipeline for {plate}")
+    output_annotated_file = str(pathlib.Path(f"{output_dir}/{plate}_sc_annotated.parquet"))
+    output_normalized_file = str(pathlib.Path(f"{output_dir}/{plate}_sc_normalized.parquet"))
+    output_feature_select_file = str(pathlib.Path(f"{output_dir}/{plate}_sc_feature_selected.parquet"))
+    output_feature_select_file = str(pathlib.Path(f"{output_dir}/{plate}_sc_feature_selected.parquet"))
```

Comment:
This is duplicated from line 89. Recommend removing.

Reply:
Great catch! will do

Reply:
removed in bc9fce06f6241ddb48fdaacf1892339d3bd1f3c6